### DISCUSSION TITLE
Обмен с 1С.

### DISCUSSION THREAD

**Discussion by LostAlly** [totumonline/totum-ru-issues-and-discussions#305] [Неопределенное]
Может кто-то реализовывал обмен с 1С Бухгалтерией?
Есть ли возможность обмена с 1С во фреше?
Только рассматриваю тотум, поэтому куча вопросов, и один из краеугольных камней это обмен с 1С.
Для начала нужна синхронизация контрагентов, номенклатуры счетов и первички, сверка финансового учета между тотум и 1с по счетам(тотум получает от банка сообщения при событии на счете, 1с по рассписанию или вручную получает выписки)
Как я себе это вижу - 
В тотум ведется операционка, 1С для целей налогообложения.
В тотму выставляются счета и отправляются в 1с, закрывающие по идее формируются в 1с и отправляются в тотум только с целью распечатки их для клиента(если клиент не использует ЭДО)

Интересно, возможно ли такое на фреше и на сколько трудозатрано.

**Comment by ttm-alexeypolunin** [comment]
За Фреш не скажу.

За коробочный 1С под столом: можно стандартными инструментами, но скилл в 1С при этом потребуется приличный тк со стороны 1С надо стартовать web-сервис и написать обработчик, который сможет отправлять то, что вам нужно на Remote в Тотум упаковывая в JSON и отвечать на запросы из Тотум.

Также 1С умеет отвечать в SOAP, для этого его нужно в этом направлении сконфигурировать.

Если вам нужно только забирать данные из 1С и вы знаете как они там организованы и 1С при этом использует postgres или mysql как БД, вы можете коннектится к их базе напрямую из Тотум — https://ru.docs.totum.online/pro-databases

Специалист 1С за средний бюджет умеющий web-обмен: https://t.me/vasdmi666

Может [Тимур Амерханов](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/231) что делал по обмену.

**Reply by k1erk** [reply]
Доброго дня. Делали с https://t.me/vasdmi666 выгрузку выписок с банка из 1С БП в тотум. Специалиста рекомендую.
![image](https://user-images.githubusercontent.com/110602346/211271018-d8be1a03-37b3-40eb-8bc5-ee68784188da.png)
![image](https://user-images.githubusercontent.com/110602346/211271122-44cee9a2-1828-4e45-b326-ffe30b6c62fd.png)

P.S. Тимуру Амерханову, еще раз спасибо за контакт.

**Comment by erpic-ru** [comment]
Оказывается, там все неплохо продвинулось, через [стандартные механизмы](https://1cfresh.com/articles/data_odata)

Я, например, загрузил контрагентов из 1С в Тотум за 15 минут, включая разбор документации.
Правда немного пришлось повозиться с русскими словами в запросах, но через функцию strUrlEncode все завелось
Двухстороннюю интеграцию рекомендую делать только в крайнем случае.

Оффтоп:
Если у вас компания на УСН, можно вообще обойтись и без всяких интеграций. Бухгалтеры с удовольствием работают в Тотуме, налоговый учет на основе выписок банка и зарплат в 1С :) Бух. и финотчетность можно в тотуме собирать без особых проблем

**Reply by LostAlly** [reply]
Прям вот то, что нужно, огромная благодарность.
В том то и проблема, у меня ИП на УСН+ ООО на ОСНО )))) проэтому весь бухучет и учет с целью налогообложения отчетность + ЭДО в 1с, а уже управленческий учет, и все остальное на стороне (сейчас вроде остановился на тотуме, точнее планируется попытка переезда на него с гуглтаблиц с изменением подхода к учету)

Спасибо про предупреждение по двухсторонней интеграции, где о проблемах можно почитать?
Обмен в любом случае будет духсторонний, только видимо придется разделить "объекты" по месту создания и одни будут синхронизироваться в одну сторону, другие в другую.

Еще раз спасибо за направление, думаю можно пометить данный ответ, как "ответ".
